enhancements/operator-dev-doc.md

```diff
@@ -0,0 +1,269 @@
+images, operand images, and others. To get a list of the components and their images that comprise a release payload, you can grab a 
+release from the [openshift release page](https://openshift-release.svc.ci.openshift.org/) and run:
+```console
+$ oc adm release info registry.svc.ci.openshift.org/ocp/release:version
+```
+
+If the above command fails, you need to authenticate against `registry.svc.ci.openshift.org`. 
+Add the necessary credentials to your local `~/.docker/config.json` (or equivalent file) like so:
+ - visit `https://api.ci.openshift.org`, `upper right corner '?'` dropdown to `Command Line Tools`
+ - copy the given `oc login https://api.ci.openshift.org --token=<hidden>`, paste in your terminal
+ - then run `oc registry login` to add your credentials to your containers tool config file     
+
+You'll notice that currently the release payload is just shy of 100 images.
+
+### CVO Cluster Operator Status
+You can check on the status of any ClusterOperator with
+```console
+$ oc get co/clusteroperatorname -o yaml
+```
+At any time the staus of a CO may be:
+- Available
+- Degraded
+- Progressing       
+
+[CVO Cluster Operator Status Conditions Overview](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md#conditions)
+
+### Metrics
+By default, an OpenShift ClusterOperator exposes [Prometheus](https://prometheus.io) metrics via `metrics` service.
+
+### Debugging
+Operators expose events that can help in debugging issues. To get operator events, run following command:
+
+```
+$ oc get events -n [cluster-operator-namespace]
+```
+
+### How do I build|update|verify|run unit-tests
+In an openshift/repo that utilizes [openshift/library-go's alpha-build-machinery](https://github.com/openshift/library-go/tree/master/alpha-build-machinery), you can get a list of all make targets with:    
+
+```console
+$ make help
+```
+
+This builds the binaries    
+```console
+$ make build
+```
+
+This runs unit tests    
+```console
+$ make test-unit
+```
+
+This will run verify-gofmt, verify-govet, verify-bindata (if applicable), verify-codegen     
+There may be other verify targets added to individual Makefiles, also.     
+```console
+$ make verify
+```
+
+This will update-bindata (if applicable), update-codegen, update-gofmt    
+```console
+$ make update
+```
+
+If you have exported your KUBECONFIG to point to a running cluster, you can run    
+the end-to-end tests that live in the repository (that would run in CI with e2e-aws-operator)    
+```console
+$ make test-e2e
+```
+
+If the repository utilizes glide for dependency management, you can update dependencies with    
+```console
+$ make update-deps
+```
+
+If the repository uses gomod for dependency management, [this doc](https://docs.google.com/document/d/1WIxyDRbXIBCihLAYRFGVkET5dM7WOQkBvnuUf2DfnL0/edit?usp=sharing) is useful.
```

Comment:
Change to `enhancements/oc/upgrade_kubernetes.md` which will be added in https://github.com/openshift/enhancements/pull/65

Reply:
ok

Reply:
added